File: app/interface/http/schemas/providers.py
"""Provider-health schemas — API Spec §9.11 (operational endpoint only).

This is the **one** place provider names legitimately appear (API Spec §8.6);
every weather-data payload stays provider-agnostic.
"""

from datetime import datetime

from app.interface.http.schemas.common import CamelModel


class ProviderHealthSchema(CamelModel):
    """API Spec §9.11."""

    provider: str
    status: str
    last_checked_at: datetime


class ProviderHealthViewSchema(CamelModel):
    """Payload of `GET /providers/health` (API Spec §8.6)."""

    providers: list[ProviderHealthSchema]

File: frontend/src/types/providers.ts
/**
 * Provider health — API Spec §9.11. **Operator-facing only.**
 *
 * This is the one payload in the contract that carries provider identity.
 * A provider name must never render on a consumer-facing screen (FDS §6.5),
 * and the endpoint requires an operator key — a consumer key gets `403`.
 */

import type { IsoDateTime } from './common'
import type { ProviderStatus } from './enums'

export interface ProviderHealth {
  /** Internal provider name. Operator screens only. */
  provider: string
  status: ProviderStatus
  lastCheckedAt: IsoDateTime
}

/** Payload of `GET /providers/health` — API Spec §8.6. */
export interface ProviderHealthView {
  providers: ProviderHealth[]
}
